addons/im_livechat/static/src/embed/common/chat_window_model_patch.js

```diff
@@ -16,6 +16,7 @@ patch(ChatWindow.prototype, {
             this.open({ notifyState: this.thread?.state !== "open" });
         } else {
             await super.close(...arguments);
+            this.thread.delete();
```

Comment:
I believe we added and removed this line several times, I'm not sure it should be there.
Are we 100% sure this will apply only to threads that are not actually existing in db?

Reply:
This line was removed when the code was moved from the chat window component to the chat window model with no real motivation. Temporary live chat threads have the same dummy id, if we do not remove them, they could interfere with each other. This will apply to every live chat thread in the embed bundle. Why should we keep outdated threads in memory? They will never be used anymore once the chat window is closed.

Reply:
But wasn't there an issue (at some point?) that the state was not reliable, and was not (or no longer) `SESSION_STATE.PERSISTED`, even for real threads?

I'm fully fine to delete for temporary, but I'm not sure if we can reliably detect them with current condition.
What is not fine is to delete for non-temporary that still exist on the server.

I thought we systematically deleted thread when necessary in other flows now.

Reply:
I can restrict to temporary threads but IMO we keep too much useless values in memory. If we know a thread won't be used anymore we should not keep it. Both persisted and temporary will pass in this branching so if you really want to only delete temporaries, I'd just add a condition on `thread.isTransient`, it works for my case.